### PR TITLE
Spikes

### PR DESCRIPTION
**Pull request by atdotde**

### CODE REVIEW

File: tests/testplan.cpp [hunk 1]
```diff
@@ -458,7 +456,7 @@ void TestPlan::testVpmbMetric45m30minTx()
 	while(!dp->minimum_gas.mbar && dp->next) dp = dp->next;
 	QCOMPARE(lrint(dp->minimum_gas.mbar / 1000.0), 108l);
 	// print first ceiling
-	printf("First ceiling %.1f m\n", (mbar_to_depth(first_ceiling_pressure.mbar, &displayed_dive) * 0.001));
+//	printf("First ceiling %.1f m\n", (mbar_to_depth(deco_state->first_ceiling_pressure.mbar, &displayed_dive) * 0.001));
 	// check benchmark run time of 141 minutes, and known Subsurface runtime of 139 minutes
```
Comment (janmulder) [inline tests/testplan.cpp]: I do not understand this change. 1) the first_ceiling_pressure is moved to the struct, so the code is corrected for that but 2) it is commented out afterwards. Comment valid for all commented lines in this file.

File: core/deco.c [hunk 2]
```diff
@@ -298,6 +296,7 @@ double tissue_tolerance_calc(const struct dive *dive, double pressure)
 		// We are doing ok if the gradient was computed within ten centimeters of the ceiling.
 		} while (fabs(ret_tolerance_limit_ambient_pressure - reference_pressure) > 0.01);
 
+		printf("initial n2 graditent 0: %f\n", deco_state->initial_n2_gradient[0]);
```
Comment (sfuchs79) [inline core/deco.c]: Wouldn't it be better to disable this debug output by default?

**Comment by dirkhh** [comment]
So this comments out a lot of debug output and then adds other debug output instead.
Hmm. That's something to talk about in the commit message in order not to confuse reviewers :-/

**Comment by atdotde** [comment]
Pushing commits in  rush is never a good idea. Will fix this.

**Comment by atdotde** [comment]
Now, that I have a little time, to reinvestigate this: first_ceiling_pressure should be in the deco_state struct (as it is a state variable of the VPMB algorithm). That struct is though not visible in the test. As the test was only printing those values but none of the tests actually used that value I decided we could do without this information rather than handing the deco_state struct up to the test. Will amend commit and commit message.

**Comment by sfuchs79** [comment]
Please be aware also about this: #601 
Means that this PR in total will make one thing better (spikes) but another worse (ceiling result in general).
Sorry that I'm of so little help here and can't do anything other than complaining ;-( 
At least I did a little bit of analysis. Hopefully Rick and you have some time to investigate.

**Comment by atdotde** [comment]
@sfuchs79 ok, but let's fix one problem at a time.

**Review by neolit123 — APPROVED** [review]

**Pull request merged**